README.md

```diff
@@ -1,4 +1,681 @@
+
+To run the example :-
+
+Compile the code.
+```sh
+	go build -o eventstreams-sdk-example
+```
+Set the required environment variables
+```sh
+# Set you API KEY.
+export API_KEY="abc123456789"
+
+# Set the Admin Endpoint to point to your cluster.
+export ADMIN_ENDPOINT="https://xyzclustername.svc01.region.eventstreams.test.cloud.ibm.com"
+
+```
+
+Run the example
+```sh
+./eventstreams-example 
+```
+
+## REST API 
+
+The following sections explain how the REST API works with examples.
+
+### Code Setup
+
+You will need to import the IBM SDK core functions and the rest API client itself.
+```golang
+	"github.com/IBM/go-sdk-core/v4/core"
+	"github.com/IBM/eventstreams-go-sdk/pkg/adminrestv1"
+```
+
+### Authentication
+
+Use one of the following methods to authenticate:
+
+- To authenticate using Basic Auth:
+  Use the user and api_key properties of the above objects as the username and password. Place these values into the Authorization header of the HTTP request in the form Basic <base64 encoding of username and password joined by a single colon (:)>.
+
+- To authenticate using a bearer token:
+  To obtain your token using the IBM Cloud CLI, first log in to IBM Cloud, then run the following command:
+  ```
+    ibmcloud iam oauth-tokens
+  ```
+    Place this token in the Authorization header of the HTTP request in the form Bearer. Both API key or JWT tokens are supported.
+
+- To authenticate directly using the api_key:
+  Place the key directly as the value of the X-Auth-Token HTTP header.
+
+```golang
+    URL := os.Getenv("ADMIN_ENDPOINT")
+	apiKey := os.Getenv("API_KEY")
+
+	if URL == "" {
+		fmt.Println("Please set env ADMIN_ENDPOINT")
+		os.Exit(1)
+	}
+
+	if apiKey == "" {
+		fmt.Println("Please set env API_KEY")
+		os.Exit(1)
+	}
+
+	basicAuthenticator, err := core.NewBasicAuthenticator("token", apiKey)
+	if err != nil {
+		fmt.Printf("Failed to Authenticate with error : %s\n", err.Error())
+		os.Exit(1)
+	}
+```
+
+
+### Create the service
+
+Create a new service object.
+
+```golang
+    // Create a new REST service.
+	ibmEventStreamsAdminRestApiService, serviceErr := ibmeventstreamsadminrestapiv1.NewIbmEventStreamsAdminRestApiV1(&ibmeventstreamsadminrestapiv1.IbmEventStreamsAdminRestApiV1Options{
+		URL:           URL,
+		Authenticator: basicAuthenticator,
+	})
+	if serviceErr != nil {
+		fmt.Println("Error Creating Service")
+		os.Exit(1)
+	}
+```
+
+### Creating a Kafka topic
+
+To get a Kafka topic detail information, the admin REST SDK issues a GET request to the `/admin/topics/TOPICNAME`
+path (where `TOPICNAME` is the name of the Kafka topic that you want to get).  
+
+Expected status codes
+  - 200: Retrieve topic details successfully in following format:
+
+
+#### Example
+
+```golang
+	// Set the retries policy.
+	serviceAPI.EnableRetries(0, 0)
+
+	// Construct an instance of the createTopicOptionsModel.
+	createTopicOptionsModel := new(ibmeventstreamsadminrestapiv1.CreateTopicOptions)
```

Comment:
Can't we use a more golang syntax like:
```golang
createTopicOptions:= &ibmeventstreamsadminrestapiv1.CreateTopicOptions{
  Name: "test-topic",
  Partitions: 2,
  ...
}
```
Do we really need all these values as pointers?
^ These 2 comments apply to all code examples

Why do we have both `Partitions` and `PartitionCount`?
Why do we have `createTopicOptionsModel.Headers = map[string]string{"x-custom-header": "x-custom-value"}`?

Reply:
If i try to set them like that i get (adminrestv1.CreateTopicOptions literal) is not a typecompiler

Reply:
and no we can't change the fact that they are all pointers as that's the way the SDK generates them, we would have to raise this as a defect with the SDK.. I think this is out of scope for me in the time I have available.